mp/src/game/server/momentum/mom_replay_entity.cpp

```diff
@@ -367,14 +369,40 @@ void CMomentumReplayGhostEntity::HandleGhostFirstPerson()
             }
         }
 
-        // interpolate vel from difference in origin
-        const Vector &pPlayerCurrentOrigin = currentStep->PlayerOrigin();
-        const Vector &pPlayerNextOrigin = nextStep->PlayerOrigin();
-        const float distX = fabs(pPlayerCurrentOrigin.x - pPlayerNextOrigin.x);
-        const float distY = fabs(pPlayerCurrentOrigin.y - pPlayerNextOrigin.y);
-        const float distZ = fabs(pPlayerCurrentOrigin.z - pPlayerNextOrigin.z);
-        const Vector interpolatedVel = Vector(distX, distY, distZ) / gpGlobals->interval_per_tick;
+
+        bool bTeleportedThisFrame = (m_cvarReplaySelection.GetInt() == 1) // Going backwards?
+            ? prevStep->Teleported() : currentStep->Teleported();
+
+        bool bTeleportedNextFrame = nextStep->Teleported();
+
+        Vector interpolatedVel;
         const float maxvel = sv_maxvelocity.GetFloat();
+        static Vector lastVel = vec3_origin;
```

Comment:
Would this method work if we had multiple replay entities? Maybe storing it as a member var would be better.

Reply:
This is true, but the absolute best way would be to store the velocity in the replay. I assumed the refactoring would do something along those lines, so I added this small hack.